website/tantalus/signals.py

```diff
@@ -0,0 +1,50 @@
+import logging
+
+from django.db.models.signals import post_save
+from django.dispatch import receiver
+
+from orders.models import Shift, Order
+from tantalus.models import TantalusProduct, TantalusOrderVenue
+from tantalus.services import sort_orders_by_product, get_tantalus_client, TantalusException
+
+
+@receiver(post_save, sender=Shift)
+def sync_to_tantalus(sender, instance, **kwargs):
+    """Synchronize Orders to Tantalus if a Shift is made finalized."""
+    if instance.finalized:
```

Comment:
This will run every time you save a shift. So it wil run twice if you are saving a shift that is finalized. Instead, you want to check the old instance here